linetools/analysis/linelimits.py

```diff
@@ -127,11 +127,15 @@ def set(self, inp):#, itype='zlim'):
                         u.dimensionless_unscaled).value - 1.
             except UnitConversionError:
                 try:
-                    self._zlim = ltu.give_dz(inp, self._z)
+                    self._zlim = ltu.give_dz(inp, self._z) + self._z
                 except ValueError:
                     raise IOError("Quantity must be length or speed")
         else:
             raise IOError("Input must be floats or Quantities")
+        # Check
+        if (self._zlim[0] > self._z) or (self._zlim[1] < self._z):
+            #import pdb; pdb.set_trace()
+            raise IOError("Invalid input.  zlim does not bound z")
```

Comment:
no
> On Aug 2, 2016, at 5:24 AM, Nicolas Tejos <notifications@github.com> wrote:
> 
> In linetools/analysis/linelimits.py <https://github.com/linetools/linetools/pull/270#discussion_r73144486>:
> 
> >                  except ValueError:
> >                      raise IOError("Quantity must be length or speed")
> >          else:
> >              raise IOError("Input must be floats or Quantities")
> > +        # Check
> > +        if (self._zlim[0] > self._z) or (self._zlim[1] < self._z):
> > +            #import pdb; pdb.set_trace()
> > +            raise IOError("Invalid input.  zlim does not bound z")
> Shouldn't it be the other way around?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/linetools/linetools/pull/270/files/f7aa3c1dd9acc1378c1f32fe38ec9b3488529323..913c296b0092120b17607f01fb677fdcff4db127#r73144486>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AD0BjbHF5XqHzOPnB_3iGkFWcge1sAMXks5qbzbhgaJpZM4JZC6f>.
> 
> !DSPAM:826,57a090e8218101135225542!

----------------------------------------------
J. Xavier Prochaska
UCO/Lick Observatory
1156 High St.
UC Santa Cruz
Santa Cruz, CA 95064

http://www.ucolick.org/~xavier/
http://imps.ucolick.org/
http://www.qpqsurvey.org/
http://k1dm3.ucolick.org/
http://www.ucolick.org/~xavier/IDL/index.html
https://github.com/pyigm/pyigm
https://github.com/linetools/linetools

831-459-2135 (Direct)
831-459-2991 (UCO/Lick Main)
831-459-5244 (Fax)